src/wrapper/jnienv.rs

```diff
@@ -330,10 +329,13 @@ impl<'a> JNIEnv<'a> {
     /// lifetime bounds since it's guaranteed to not get GC'd by java. It
     /// releases the GC pin upon being dropped.
     pub fn new_global_ref(&self, obj: JObject) -> Result<GlobalRef> {
-        non_null!(obj, "new_global_ref obj argument");
-        let new_ref: JObject = jni_non_null_call!(self.internal, NewGlobalRef, obj.into_inner());
-        let global = unsafe { GlobalRef::from_raw(self.get_java_vm()?, new_ref.into_inner()) };
-        Ok(global)
+        let new_ref: JObject = jni_unchecked!(self.internal, NewGlobalRef, obj.into_inner()).into();
+        if !new_ref.is_null() {
+            let global = unsafe { GlobalRef::from_raw(self.get_java_vm()?, new_ref.into_inner()) };
+            Ok(global)
+        } else {
+            Err(ErrorKind::NullPtr("new_global_ref").into())
```

Comment:
This seems like a more verbose version of the previous code that checked the argument for null. The spec allows to "create" and "delete" `null` global refs (it won't create anything, but return a `null` from new and do nothing on delete) — https://docs.oracle.com/en/java/javase/11/docs/specs/jni/functions.html#newglobalref

If some method returns a reference that may be `null`, and a user needs to store that nullable reference in a `GlobalRef`, they will get an error instead.

Reply:
Basically, that's the same question of how to deal with `null`s consistently: #148 

I’d go with a symmetric solution to local references, and reconsider in #148